app/templates/courses/teacher-class-view.jade

```diff
@@ -201,7 +202,7 @@ mixin studentRow(student)
   tr.student-row.alternating-background
     td.checkbox-col.student-checkbox
       .checkbox-flat
-        input(type='checkbox' id='checkbox-student-' + student.id, data-student-id=student.id)
+        input(type='checkbox' id='checkbox-student-' + student.id, data-student-id=student.id, checked=state.get('checkboxStates')[student.id])
```

Comment:
Might want to pull `checkboxStates` out outside the loop, just so it doesn't called once per student? Probably not actually a big hit on performance.